### PR TITLE
Feature/cryp 50 implement an endpoint to return the list of offers received by artist

### PR DESCRIPTION
**Pull request by Afaqsoftaxus**
## Jira

 https://softaxus-jira.atlassian.net/browse/CRYP-50

## Changes

• Added a new endpoint to get the list of offers Received by the artist
• Added the respective handler
• Added unit tests for the handler

## Testing

 https://softaxus-jira.atlassian.net/browse/CRYP-50?focusedCommentId=10183

### CODE REVIEW

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 1]
```diff
@@ -15,7 +13,7 @@ public ArtistController(IMediator mediator, ILogger<ArtistController> logger) :
     }
 
     /// <summary>
-    /// Returns artist an by searched by id.
+    /// Returns an artist  by searched by id.
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:16]: Spacing

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 2]
```diff
@@ -59,7 +57,7 @@ public ArtistController(IMediator mediator, ILogger<ArtistController> logger) :
     /// <param name="id" example="507f191e810c19729de860ea"></param>
     /// <response code="204">Delete Artist by Id</response>
     /// <returns></returns>
-    [HttpDelete(template: "{id:regex(^[[A-Za-z0-9]]*$):required}", Name = "DeleteArtistById", Order = 2)]
+    [HttpDelete(template: "{id:regex(^[[A-Za-z0-9]]*$):required}", Name = "DeleteArtistById", Order = 3)]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:60]: Order should be 2

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 3]
```diff
@@ -81,7 +79,7 @@ public async Task<IActionResult> DeleteArtistById([FromRoute] string id)
     /// </summary>
     /// <param name="artist"></param>
     /// <returns></returns>
-    [HttpPost(Name = "PostArtist", Order = 3)]
+    [HttpPost(Name = "PostArtist", Order = 4)]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:82]: Order is wrong

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 4]
```diff
@@ -105,12 +103,12 @@ public async Task<IActionResult> PostArtist([FromBody] CreateArtistDto artist)
     /// <response code="404">Not Found response with 404 code and information message</response>
     /// <response code="400">Bad Request response with 400 code and information message</response>
     /// <returns></returns>
-    [HttpGet("{userId:int:required}/userId", Name = "GetArtistByUserId", Order = 4)]
+    [HttpGet("{userId:int:required}/userId", Name = "GetArtistByUserId", Order = 5)]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:106]: Order

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 5]
```diff
@@ -105,12 +103,12 @@ public async Task<IActionResult> PostArtist([FromBody] CreateArtistDto artist)
...
     [RequiresParameter(Name = "userId", Required = true, Source = OpenApiParameterLocation.Path, Type = typeof(int))]
     [SwaggerRequestExample(typeof(GetArtistByUserIdRequest), typeof(GetArtistByUserIdRequestExample))]
     [ProducesResponseType(typeof(GetArtistByUserIdResponse), (int)HttpStatusCode.OK)]
-    [ProducesResponseType(typeof(NotFoundArtistByWalletAddressResponse), (int)HttpStatusCode.NotFound)]
-    [ProducesResponseType(typeof(BadRequestArtistByUserIdResponse), (int)HttpStatusCode.BadRequest)]
+    [ProducesResponseType(typeof(NotFoundArtistByUserIdResponse), (int)HttpStatusCode.NotFound)]
+    [ProducesResponseType(typeof(NotFoundArtistByUserIdResponse), (int)HttpStatusCode.BadRequest)]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:111]: Wrong response type class

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 6]
```diff
@@ -141,7 +139,7 @@ public async Task<IActionResult> GetArtistByUserId([FromRoute] int userId, [From
     /// <response code="404">Not Found response with 404 code and information message</response>
     /// <response code="400">Bad Request response with 400 code and information message</response>
     /// <returns></returns>
-    [HttpPatch("{userId:int:required}", Name = "PatchArtist", Order = 5)]
+    [HttpPatch("{userId:int:required}", Name = "PatchArtist", Order = 6)]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:142]: Order

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 7]
```diff
@@ -160,6 +158,51 @@ public async Task<IActionResult> PatchArtist([FromRoute] int userId, [FromBody]
         };
     }
 
+    /// <summary>
+    /// Returns Offers  Recived By Artist
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:162]: Spelling and spacing issue

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 8]
```diff
@@ -160,6 +158,51 @@ public async Task<IActionResult> PatchArtist([FromRoute] int userId, [FromBody]
...
+    /// </summary>
+    /// <param name="userId" example="0x507f191e810c19729de860ea"></param>
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:164]: Wrong example for userId

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 9]
```diff
@@ -160,6 +158,51 @@ public async Task<IActionResult> PatchArtist([FromRoute] int userId, [FromBody]
...
+    /// <param name="paginationParameters"></param>
+    /// <param name="mediaType" example="application/json"></param>
+    /// <param name="cancellationToken" example="default"></param>
+    /// <response code="200">Success response with 200 code and information message about record found</response>
+    /// <response code="404">Not Found response with 404 code and information message</response>
+    /// <response code="400">Bad Request response with 400 code and information message</response>
+    /// <returns></returns>
+    [HttpGet("{userId:int:required}/offersRecived", Name = "GetOffersReceivedByArtist", Order = 7)]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:172]: Endpoint Url contains spelling mistake.

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 10]
```diff
@@ -160,6 +158,51 @@ public async Task<IActionResult> PatchArtist([FromRoute] int userId, [FromBody]
...
+    [RequiresParameter(Name = "userId", Required = true, Source = OpenApiParameterLocation.Path, Type = typeof(int))]
+    [RequiresParameter(Name = "paginationParameters", Required = true, Source = OpenApiParameterLocation.Query, Type = typeof(PaginationParameters))]
+    [RequiresParameter(Name = "mediaType", Required = true, Source = OpenApiParameterLocation.Header, Type = typeof(string))]
+    [SwaggerRequestExample(typeof(GetOffersRecivedByArtistRequest), typeof(GetOffersRecivedByArtistRequestExample))]
+    [ProducesResponseType(typeof(GetOffersRecivedByArtistResponse), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(NotFoundGetOffersRecivedByArtistResponse), (int)HttpStatusCode.NotFound)]
+    [ProducesResponseType(typeof(BadRequestGetOffersRecivedByArtistResponse), (int)HttpStatusCode.BadRequest)]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:179]: Spelling mistake

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 11]
```diff
@@ -171,7 +214,7 @@ public async Task<IActionResult> PatchArtist([FromRoute] int userId, [FromBody]
     /// <response code="404">Not Found response with 404 code and information message</response>
     /// <response code="400">Bad Request response with 400 code and information message</response>
     /// <returns></returns>
-    [HttpGet("{userId:int:required}/offersMade", Name = "OffersMadeByArtist", Order = 6)]
+    [HttpGet("{userId:int:required}/offersMade", Name = "OffersMadeByArtist", Order = 8)]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:217]: Order

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 12]
```diff
@@ -262,6 +305,20 @@ private IReadOnlyList<Links> CreateArtistLinks(string? id = null, int? userId =
...
+        return links.AsReadOnly();
+    }
+
+    private IReadOnlyList<Links> CreateOfferLinks(OffersDto dto, string? field)
+    {
+        Links link;
+        List<Links> links = new List<Links>();
+
         return links.AsReadOnly();
     }
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:323]: Wrong implementation. This is just returning an empty list of links to the client end.

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponse.cs [hunk 13]
```diff
@@ -0,0 +1,19 @@
+﻿namespace CryptoAxus.Application.Features.Artist.GetOffersRecivedByArtist.Response;
+
+public class GetOffersRecivedByArtistResponse : PaginationResponse<List<OffersDto>>
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponse.cs:3]: Should be Offers Dto with links.

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponseExample.cs [hunk 14]
```diff
@@ -0,0 +1,36 @@
+﻿namespace CryptoAxus.Application.Features.Artist.GetOffersRecivedByArtist.Response;
+
+public class GetOffersRecivedByArtistResponseExample : IExamplesProvider<GetOffersRecivedByArtistResponse>
+{
+    public GetOffersRecivedByArtistResponse GetExamples()
+    { 
+            return new GetOffersRecivedByArtistResponse(HttpStatusCode.OK, "Records found successfully.", new List<OffersDto>
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponseExample.cs:7]: Should be Offers Dto with links.

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponseExample.cs [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+         {
+            new OffersDto(ObjectId.GenerateNewId(),
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponseExample.cs:9]: Should be Offers Dto with links.

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/NotFoundGetOffersRecivedByArtistResponse.cs [hunk 16]
```diff
@@ -0,0 +1,8 @@
+﻿namespace CryptoAxus.Application.Features.Artist.GetOffersRecivedByArtist.Response;
+
+public class NotFoundGetOffersRecivedByArtistResponse : PaginationResponse<List<OffersDto>>
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/NotFoundGetOffersRecivedByArtistResponse.cs:3]: Should be Offers Dto with links.

File: src/CryptoAxus.Domain/Collections/OffersDocument .cs [hunk 17]
```diff
@@ -38,6 +38,7 @@ public OffersDocument() : base()
                           ObjectId createdBy,
                           DateTime? lastModifiedDate = null,
                           ObjectId? lastModifiedBy = null,
+                          string? signature = null,
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Domain/Collections/OffersDocument .cs:41]: This field was removed in the previous ticket.

**Review by junaid-softaxus — CHANGES_REQUESTED** [review]
Address the changes requested in the PR review. Also add the test case to verify the functionality of the BadRequest scenario.

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 3 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:82]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 2 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:60]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 4 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:106]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 6 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:142]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 8 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:164]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 9 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:172]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 10 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:179]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 11 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:217]: Resolved

File: src/CryptoAxus.Domain/Collections/OffersDocument .cs (hunk 17 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Domain/Collections/OffersDocument .cs:41]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponseExample.cs (hunk 14 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponseExample.cs:7]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/NotFoundGetOffersRecivedByArtistResponse.cs (hunk 16 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/NotFoundGetOffersRecivedByArtistResponse.cs:3]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 12 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:323]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponseExample.cs (hunk 15 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersRecivedByArtistResponseExample.cs:9]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 18]
```diff
@@ -171,11 +218,11 @@ public async Task<IActionResult> PatchArtist([FromRoute] int userId, [FromBody]
     /// <response code="404">Not Found response with 404 code and information message</response>
     /// <response code="400">Bad Request response with 400 code and information message</response>
     /// <returns></returns>
-    [HttpGet("{userId:int:required}/offersMade", Name = "OffersMadeByArtist", Order = 6)]
+    [HttpGet("{userId:int:required}/offersMade", Name = "OffersMadeByArtist", Order = 7)]
     [RequiresParameter(Name = "userId", Required = true, Source = OpenApiParameterLocation.Path, Type = typeof(int))]
     [RequiresParameter(Name = "paginationParameters", Required = true, Source = OpenApiParameterLocation.Query, Type = typeof(PaginationParameters))]
     [RequiresParameter(Name = "mediaType", Required = true, Source = OpenApiParameterLocation.Header, Type = typeof(string))]
-    [SwaggerRequestExample(typeof(GetOffersMadeByArtistRequest), typeof(GetOffersReceivedByArtistRequestExample))]
+    [SwaggerRequestExample(typeof(GetOffersMadeByArtistRequest), typeof(GetOffersMadeByArtistResponseExample))]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:225]: This should be `GetOffersMadeByArtistRequestExample`

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 19]
```diff
@@ -262,6 +309,37 @@ private IReadOnlyList<Links> CreateArtistLinks(string? id = null, int? userId =
...
+        {
+            link = new Links(Url.RouteUrl("GetArtistOffersReceivedByArtist", new { userId }),
+                             Constants.SelfRel,
+                             Constants.GetMethod);
+            link.Href = link.Href?.Replace(Constants.ApiValue,
+                                           $"{HttpContext?.Request.Scheme}://{HttpContext?.Request.Host}{Constants.ApiValue}");
+            links.Add(link);
+        }
         return links.AsReadOnly();
     }
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:344]: Wrong implementation. Should be related to the offers that are returned in the response.

File: src/CryptoAxus.Application/Features/Artist/GetOffersMadeByArtist/Request/GetOffersMadeByArtistRequestExample.cs [hunk 20]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace CryptoAxus.Application.Features.Artist.GetOffersMadeByArtist.Request;
 
-public class GetOffersReceivedByArtistRequestExample : IExamplesProvider<GetOffersMadeByArtistRequest>
+public class GetOffersMadeByArtistRequestExample : IExamplesProvider<GetOffersMadeByArtistRequest>
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersMadeByArtist/Request/GetOffersMadeByArtistRequestExample.cs:3]: Why is this file changed?

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Handler/GetOffersReceivedByArtistHandler.cs [hunk 21]
```diff
@@ -0,0 +1,51 @@
+﻿namespace CryptoAxus.Application.Features.Artist.GetOffersReceivedByArtist.Handler;
+
+public class GetOffersReceivedByArtistHandler : BaseHandler<GetOffersReceivedByArtistHandler>, 
+                                               IRequestHandler<GetOffersReceivedByArtistRequest, GetOffersReceivedByArtistResponse>
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Handler/GetOffersReceivedByArtistHandler.cs:4]: Alignment.

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Handler/GetOffersReceivedByArtistHandler.cs [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+        _repository = repository;
+        _cacheService = cacheService;
+    }
+
+    public async Task<GetOffersReceivedByArtistResponse> Handle(GetOffersReceivedByArtistRequest request,
+                                                                CancellationToken cancellationToken = default)
+    {
+        GetOffersReceivedByArtistResponse? response =
+                await _cacheService.GetAsync<GetOffersReceivedByArtistResponse>(
+                key: $"offersRecivedByArtistP{request.PaginationParameters.PageNumber}S{request.PaginationParameters.PageSize}");
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Handler/GetOffersReceivedByArtistHandler.cs:20]: Spelling mistake.

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Handler/GetOffersReceivedByArtistHandler.cs [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+        PaginationData paginationData = new PaginationData(countTask.Result,
+                                                           request.PaginationParameters.PageNumber,
+                                                           request.PaginationParameters.PageSize);
+
+        response = new GetOffersReceivedByArtistResponse(HttpStatusCode.OK, "Records found successfully.",
+                                                         offersTask.Result.Adapt<List<OffersDtoWithLinks>>(),
+                                                         paginationData);
+
+        await _cacheService.SetAsync(key: $"offersRecivedByArtistP{request.PaginationParameters.PageNumber}S{request.PaginationParameters.PageSize}",
+                                     value: JsonSerializer.Serialize(response));
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Handler/GetOffersReceivedByArtistHandler.cs:47]: Spelling mistake.

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersReceivedByArtistResponseExample.cs [hunk 24]
```diff
@@ -0,0 +1,38 @@
+﻿namespace CryptoAxus.Application.Features.Artist.GetOffersReceivedByArtist.Response;
+
+public class GetOffersReceivedByArtistResponseExample : IExamplesProvider<GetOffersReceivedByArtistResponse>
+{
+    public GetOffersReceivedByArtistResponse GetExamples()
+    { 
+            return new GetOffersReceivedByArtistResponse(HttpStatusCode.OK, "Records found successfully.", new List<OffersDtoWithLinks>
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersReceivedByArtistResponseExample.cs:7]: Alignment.

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersReceivedByArtistResponseExample.cs [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+                                   286,
+                                   3,
+                                   2,
+                                   DateTime.UtcNow.AddHours(6),
+                                   null,
+                                   ObjectId.GenerateNewId(),
+                                   DateTime.UtcNow.AddDays(-5),
+                                   ObjectId.GenerateNewId(),
+                                   false)
+        }, new PaginationData(5, 1, 2, "http://cryptoaxus.com/api/artist/5428/offersMade"));
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersReceivedByArtistResponseExample.cs:36]: Incorrect link provided in the example.

**Review by junaid-softaxus — CHANGES_REQUESTED** [review]
Feedback provided.

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Handler/GetOffersReceivedByArtistHandler.cs (hunk 23 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Handler/GetOffersReceivedByArtistHandler.cs:47]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersReceivedByArtistResponseExample.cs (hunk 24 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetOffersRecivedByArtist/Response/GetOffersReceivedByArtistResponseExample.cs:7]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetOffersMadeByArtist/Request/GetOffersMadeByArtistRequestExample.cs (hunk 20 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetOffersMadeByArtist/Request/GetOffersMadeByArtistRequestExample.cs:3]: This was not changed on my end as, I have send the screenshot on slack

**Comment by Afaqsoftaxus** [comment]
> Feedback provided.

All the changes addressed.

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 26]
```diff
@@ -262,7 +309,66 @@ private IReadOnlyList<Links> CreateArtistLinks(string? id = null, int? userId =
...
+        links.Add(link);
+
+        link = new Links(href: Url.RouteUrl("AddOffersReceivedByArtist", new { userId }),
+                         "post_offers",
+                         Constants.PostMethod);
+        link.Href = link.Href?.Replace(Constants.ApiValue,
+                                       $"{HttpContext?.Request.Scheme}://{HttpContext?.Request.Host}{Constants.ApiValue}");
+        links.Add(link);
         return links.AsReadOnly();
+
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:371]: Links are not correct.

**Review by junaid-softaxus — CHANGES_REQUESTED** [review]
Please update the links method correctly.

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 19 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 26 above)
Comment (junaid-softaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs]: Links were updated correctly.

**Review by junaid-softaxus — APPROVED** [review]
